### PR TITLE
UDP Message passing

### PR DESCRIPTION
**Pull request by ganeshram93**
UDP Message-passing of Heart Rate Sensor data from Sensor Node to Border Router.

### CODE REVIEW

File: cli_ftd/main.c [hunk 1]
```diff
@@ -120,6 +122,8 @@ int main(void) {
 
   cli_taskCreate();
 
+  udp_comm_taskCreate();
```
Comment (coder137) [inline cli_ftd/main.c:125]: Isnt this also CC1352R1 specific ?

File: cli_ftd/task_config.h [hunk 2]
```diff
@@ -69,7 +69,7 @@
  * Priority of the Application task.
  */
 #ifndef TASK_CONFIG_CLI_TASK_PRIORITY
-#define TASK_CONFIG_CLI_TASK_PRIORITY 3
+#define TASK_CONFIG_CLI_TASK_PRIORITY 4
```
Comment (coder137) [inline cli_ftd/task_config.h:72]: Shift this to your .c udp files
Make it internal static const

File: cli_ftd/user/heartbeat/heartbeat.c [hunk 3]
```diff
@@ -128,13 +139,16 @@ void *heartbeat_task(void *arg0) {
     printf("LED Toggle %ld \r\n", count++);
 #endif
     GPIO_toggle(CONFIG_GPIO_GLED);
-
     body = readBpm();
+
+#if HEARTBEAT_DEBUG == 1
```
Comment (coder137) [inline cli_ftd/user/heartbeat/heartbeat.c:144]: Use the debugprintf macro instead
See stm32 module for the usage

File: cli_ftd/user/heartbeat/heartbeat.h [hunk 4]
```diff
@@ -5,5 +5,5 @@
  * Creation funciton for the heartbeat application task.
  */
 extern void heartbeat_taskCreate(void);
-
+extern void get_heartbeat_data(char *temp_buff);
```
Comment (coder137) [inline cli_ftd/user/heartbeat/heartbeat.h:8]: Why are these extern?
Also use preet's naming convention
`heartbeat__get_data` or `heartbeat_get_data`

File: cli_ftd/user/udp_communication/udp_comm.c [hunk 5]
```diff
@@ -0,0 +1,139 @@
...
+
+  while (1) {
+    if (!GPIO_read(CONFIG_GPIO_BTN1)) {
+      OtRtosApi_lock();
+      snprintf(udp_final_message, sizeof(udp_final_message), "RLOC16:%04x,",
+               otThreadGetRloc16(aInstance));
+      OtRtosApi_unlock();
+      get_heartbeat_data(udp_temp_msg_buff);
+      strcat(udp_final_message, udp_temp_msg_buff);
+      printf("BTN-1 pressed: Sent the Msg: %s\r\n", udp_final_message);
```
Comment (coder137) [inline cli_ftd/user/udp_communication/udp_comm.c:100]: So data is sent only on button press, it does not get sent periodically?

**Review by coder137 — CHANGES_REQUESTED** [review]

File: cli_ftd/main.c (hunk 1 above)
Comment (ganeshram93) [reply cli_ftd/main.c:125]: No, this is platform-independent.

File: cli_ftd/task_config.h (hunk 2 above)
Comment (ganeshram93) [reply cli_ftd/task_config.h:72]: This is CLI task specific and TI's CLI code.

File: cli_ftd/user/heartbeat/heartbeat.h (hunk 4 above)
Comment (ganeshram93) [reply cli_ftd/user/heartbeat/heartbeat.h:8]: Ok, will address.

File: cli_ftd/user/udp_communication/udp_comm.c (hunk 5 above)
Comment (ganeshram93) [reply cli_ftd/user/udp_communication/udp_comm.c:100]: Yes for now it is implemented that way. I can remove the dependency of the button.

File: cli_ftd/user/heartbeat/heartbeat.c (hunk 3 above)
Comment (ganeshram93) [reply cli_ftd/user/heartbeat/heartbeat.c:144]: Ok, will address.

File: cli_ftd/main.c (hunk 1 above)
Comment (ganeshram93) [reply cli_ftd/main.c:125]: I will modify CMakeLists.txt to include it as common platform target.

**Review by nikhilpeoria — APPROVED** [review]

File: cli_ftd/CMakeLists.txt [hunk 6]
```diff
@@ -3,8 +3,8 @@ cmake_minimum_required(VERSION 3.17)
 # User Defines
 
 # ! IMPORTANT, Toggle only between these two options
-# set(TARGET_BOARD "CC1352P_2_LAUNCHXL")
-set(TARGET_BOARD "CC1352R1_LAUNCHXL")
+set(TARGET_BOARD "CC1352P_2_LAUNCHXL")
+# set(TARGET_BOARD "CC1352R1_LAUNCHXL")
```
Comment (coder137) [inline cli_ftd/CMakeLists.txt:7]: change this back to CC1352R1

**Review by coder137 — CHANGES_REQUESTED** [review]

File: cli_ftd/CMakeLists.txt (hunk 6 above)
Comment (ganeshram93) [reply cli_ftd/CMakeLists.txt:7]: Will address.

**Review by coder137 — APPROVED** [review]

**Pull request merged**